src/selectors/entities/channels.js

```diff
@@ -575,3 +577,16 @@ export const getSortedDirectChannelIds = createIdsSelector(
         return directChannels.map((c) => c.id);
     }
 );
+
+export const getGroupOrDirectChannelVisibility = createSelector(
+    getChannel,
+    getMyChannelMemberships,
+    getConfig,
+    getMyPreferences,
+    getCurrentUser,
+    getUsers,
+    getLastPostPerChannel,
+    (channel, memberships, config, myPreferences, currentUser, users, lastPosts) => {
+        return isGroupOrDirectChannelVisible(channel, memberships, config, myPreferences, currentUser.id, users, lastPosts);
+    }
+);
```

Comment:
Updating the webapp to use this instead of calling `isXVisible` directly. Simplifies those call sites a lot.

Reply:
Actually, can you either make this into a `makeGetGroupOrDirectChannelVisibility` that returns a selector or just make it a plain function? Since it takes an argument besides the state, memoization is impossible with a single instance if there's multiple places calling this, so I'd rather either make it a factory (so it can be memoized if necessary) or not have it be a selector at all so there's no illusion that it would be memoized well.